packages/appeals-service-api/src/routes/v2/appeal-cases/_caseReference/events/event.spec.js

```diff
@@ -0,0 +1,198 @@
+const supertest = require('supertest');
+const app = require('../../../../../app');
+const { createPrismaClient } = require('../../../../../db/db-client');
+const { seedStaticData } = require('@pins/database/src/seed/data-static');
+const { APPEAL_USER_ROLES } = require('@pins/common/src/constants');
+const crypto = require('crypto');
+
+const { isFeatureActive } = require('../../../../../configuration/featureFlag');
+
+/** @type {import('@prisma/client').PrismaClient} */
+let sqlClient;
+/** @type {import('supertest').SuperTest<import('supertest').Test>} */
+let appealsApi;
+
+let validUser;
+const validLpa = 'Q9999';
+
+jest.mock('../../../../../configuration/featureFlag');
+jest.mock('../../../../../../src/services/object-store');
+jest.mock('express-oauth2-jwt-bearer', () => {
+	let currentSub = '';
+
+	return {
+		auth: jest.fn(() => {
+			return (req, _res, next) => {
+				req.auth = {
+					payload: {
+						sub: currentSub
+					}
+				};
+				next();
+			};
+		}),
+		setCurrentSub: (newSub) => {
+			currentSub = newSub;
+		}
+	};
+});
+
+jest.setTimeout(30000);
+
+beforeAll(async () => {
+	///////////////////////////////
+	///// SETUP TEST DATABASE ////
+	/////////////////////////////
+	sqlClient = createPrismaClient();
+
+	/////////////////////
+	///// SETUP APP ////
+	///////////////////
+	appealsApi = supertest(app);
+
+	await seedStaticData(sqlClient);
+
+	const user = await sqlClient.appealUser.create({
+		data: {
+			email: crypto.randomUUID() + '@example.com'
+		}
+	});
+	validUser = user.id;
+});
+
+beforeEach(async () => {
+	// turn all feature flags on
+	isFeatureActive.mockImplementation(() => {
+		return true;
+	});
+});
+
+afterEach(async () => {
+	jest.clearAllMocks();
+});
+
+afterAll(async () => {
+	await sqlClient.$disconnect();
+});
+
+/**
+ * @returns {Promise<string>}
+ */
+const createAppeal = async (caseRef) => {
+	const today = new Date();
+
+	let futureEndDate = new Date();
+	futureEndDate.setDate(today.getDate() + 1);
+
+	let pastEndDate = new Date();
+	pastEndDate.setDate(today.getDate() - 1);
+
+	const appeal = await sqlClient.appeal.create({
+		include: {
+			AppealCase: true
+		},
+		data: {
+			Users: {
+				create: {
+					userId: validUser,
+					role: APPEAL_USER_ROLES.APPELLANT
+				}
+			},
+			AppealCase: {
+				create: {
+					caseReference: caseRef,
+					LPACode: validLpa,
+					LPAName: 'Test',
+					appealTypeCode: 'HAS',
+					appealTypeName: 'HAS1',
+					decision: 'refused',
+					originalCaseDecisionDate: new Date(),
+					costsAppliedForIndicator: true,
+					LPAApplicationReference: 'abc',
+					siteAddressLine1: 'abcd',
+					siteAddressPostcode: 'abc123'
+				}
+			}
+		}
+	});
+
+	await sqlClient.event.createMany({
+		data: [
+			{
+				appealCaseReference: caseRef,
```

Comment:
should this key be caseReference instead?

Reply:
yes, yes it should